src/routes/docs/tutorials/apple/step-4/+page.markdoc

```diff
@@ -0,0 +1,193 @@
+step: 4
+---
+
+# Appwrite service {% #appwrite-service %}
+
+Create a new file AppwriteService.swift inside **Shared/Services** folder and add the following code to it, replacing [YOUR_PROJECT_ID] with your project ID.
+
+```swift
+import Foundation
+import Appwrite
+import JSONCodable
+
+class AppwriteService {
+    var client: Client
+    var account: Account
+    var database: Databases
+    
+    public init() {
+        self.client = Client()
+            .setEndpoint("https://cloud.appwrite.io/v1")
+            .setProject("[YOUR_PROJECT_ID]")
+        self.account = Account(client)
+        self.database = Databases(client)
+    }
+    
+    
+    public func currentSession() async throws -> User<[String: AnyCodable]> {
+        try await account.get()
+    }
+    
+    public func onRegister(
+        _ email: String,
+        _ password: String
+    ) async throws -> User<[String: AnyCodable]> {
+        try await account.create(
+            userId: ID.unique(),
+            email: email,
+            password: password
+        )
+    }
+    
+    public func onLogin(
+        _ email: String,
+        _ password: String
+    ) async throws -> Session {
+        try await account.createEmailSession(
+            email: email,
+            password: password
+        )
+    }
+    
+    public func onLogout() async throws {
+        _ = try await account.deleteSession(
+            sessionId: "current"
+        )
+    }
+    
+    
+    public func onAccountDelete() async throws {
+        _ = try await account.deleteSession(
+            sessionId: "current"
+        )
+    }
+    
+    public func getDocs<T>(_ db: Database, _ collection: DatabaseCollections, queries: [String]? = nil) async throws -> DocumentList<T> {
+        try await database.listDocuments<T>(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            queries: queries,
+            nestedType: T.self
+        )
+    }
+    
+    public func insertDoc(_ db: Database, _ collection: DatabaseCollections, data: Any) async throws {
+        _ = try await database.createDocument(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            documentId: ID.unique(),
+            data: data
+        )
+    }
+    
+    
+    public func updateDoc(_ db: Database, _ collection: DatabaseCollections, _ id: String, data: Any) async throws {
+        _ = try await database.updateDocument(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            documentId: id,
+            data: data
+        )
+    }
+    
+    
+    public func removeDoc(_ db: Database, _ collection: DatabaseCollections, _ id: String, data: Any) async throws {
+        _ = try await database.deleteDocument(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            documentId: id
+        )
```

Comment:
<img width="1125" alt="Screenshot 2023-10-24 at 6 28 57 PM" src="https://github.com/appwrite/website/assets/29069505/062a37fc-39a2-41e8-b777-d9d07b9f4c59">
Is there a chance your using an old XCode/swift version that could cause inconsistencies like this? XCode didn't like your code here :P

Reply:
I'm using Xcode 15, as shown in the screenshot below.

<img width="1390" alt="Screenshot 2023-10-25 at 09 08 58" src="https://github.com/appwrite/website/assets/41507005/752789c6-3792-4c71-a990-05b732d93878">
<img width="535" alt="Screenshot 2023-10-25 at 09 09 08" src="https://github.com/appwrite/website/assets/41507005/ffb4e4fe-009e-4227-92bb-2ee841aaec2d">

Reply:
I just noticed what likely happened; a step is missing. Sorry about that. I have updated step 3 to include that part